### PR TITLE
Add instructions to revert a release

### PR DESCRIPTION
**Pull request by timothybest-usds**
## Related Issue or Background Info

We have been operating under a fix and move forward model for buggy releases. In preparation for rolling out [an on-call schedule](https://docs.google.com/document/d/1FU3a-No7Pv5wmRGE1uwAly_3oWHaggkiIaNheiqF2XU/edit?usp=sharing), to increase the stability of our application and to ensure no one has to "fix and move forward" at 3am we should move to rolling back buggy releases and fixing in a non-prod env

## Changes Proposed

- Add instructions for rolling back to a previous version

## Additional Information
- It seems that the create release form doesn't seem to let you target a tag or arbitrary commit (only allows "recent" commits)
- To test the purposed flow I checked out `v29`, created and published a branch and confirmed that the branch is select able in the create release form. 
- Would like to test this process out during business hours with a trivial change

## Screenshots / Demos
None

### UI
- [x] Any changes to the UI/UX have been approved by design 
- [x] Any new or edited error messages have been approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support

### CODE REVIEW

**Review by benwarfield-usds — CHANGES_REQUESTED** [review]
We should include the much-simpler process of rolling back using the staging slot.

**Comment by timothybest-usds** [comment]
> We should include the much-simpler process of rolling back using the staging slot.

My understanding is that the staging slot will only work for the backend. We would still need to redeploy the frontend. If that is true then I would like to go forward with this strategy and we can follow up with a V2 to use the staging slot and redeploy the frontend

**Comment by benwarfield-usds** [comment]
Solid point about the front end—we talked about that issue but didn't conclude anything very useful. I think there should be a way to snap back the front end as well, but there isn't now. (And staging slots wouldn't be it.)

File: README.md [hunk 1]
```diff
@@ -325,6 +326,26 @@ Navigate to [New Release Form](https://github.com/CDCgov/prime-simplereport/rele
 4. Click publish release
 5. Verify the changes are live by ensuring the deployed commit hash matches the commit hash on the release. This is done my going to `/app/static/commit.txt` and `/api/actuator/info`
 
+### Revert to a Previous Release
+
+1. Find the version tag for the release you want to revert to.
+2. Checkout that version and create a new branch
+    ```bash
+    $ git checkout
```
Comment (benwarfield-usds) [inline README.md:334]: Should this be `git checkout ${version_tag}`?

File: README.md [hunk 2]
```diff
@@ -325,6 +326,26 @@ Navigate to [New Release Form](https://github.com/CDCgov/prime-simplereport/rele
...
+     ```bash
+    $ git checkout -b revert-to-${version_tag} && git push -u
+    ```
+4. Navigate to [New Release Form](https://github.com/CDCgov/prime-simplereport/releases/new) page
+5. Add a version tag: `revert-to-${version_tag}`.
+    - If a version has already been reverted to in the past and needs to be again add an counter to the tag: `revert-to-${version_tag}-${X}`
+6. Add a release title `Revert to ${version_tag}`
+7. Add a description briefly explaining why the revert is needed
+8. Click publish release
+9. Verify the changes are live by ensuring the deployed commit hash matches the commit hash on the release. This is done my going to `/app/static/commit.txt` and `/api/actuator/info`
```
Comment (benwarfield-usds) [inline README.md:346]:
```suggestion
9. Verify the changes are live by ensuring the deployed commit hash matches the commit hash on the release. This is done by going to `/app/static/commit.txt` and `/api/actuator/info`
```

**Review by benwarfield-usds — APPROVED** [review]
I feel like we can improve on that process, but having a process we can iterate on is good.

**Comment by timothybest-usds** [comment]
> Solid point about the front end—we talked about that issue but didn't conclude anything very useful. I think there should be a way to snap back the front end as well, but there isn't now. (And staging slots wouldn't be it.)

Currently I think we arn't cleaning up any of the static resources. It may be possible to change just the `index.html` file back to the old version

File: README.md (hunk 1 above)
Comment (timothybest-usds) [reply README.md:334]: Yest it should!
```suggestion
    $ git checkout ${version_tag}
```

**Review by jdorothy — APPROVED** [review]
This looks good to me. I agree that we can probably tweak this as far as Azure slots / updating static files, but I also like that anyone can follow this procedure even if they don't have Azure access.

**Pull request merged**